app/models/comment.rb

```diff
@@ -25,4 +27,16 @@ def self.attributes
       'updated_at'
     ]
   end
+
+  # All of the users who should be notified when a comment is created
+  def listeners
+    users_to_notify = []
+    users_to_notify += self.proposal.currently_awaiting_approvers
+    users_to_notify += self.proposal.approvals.approved.map(&:user)
+    users_to_notify += self.proposal.observers
+    users_to_notify << self.proposal.requester
```

Comment:
What do you think about sharing some logic with `Proposal#users`? Also, there *shouldn't* be duplicates, but I'd make `users_to_notify` a [`Set`](http://ruby-doc.org/stdlib-2.2.2/libdoc/set/rdoc/Set.html) anyway.